### ISSUE TITLE
Group tasks that can be done in parallel after submitting.

### ISSUE THREAD

**Issue by joeribekker** [open-formulieren/open-forms#669]
In https://github.com/open-formulieren/open-forms/pull/664/files#diff-6a77b95a156d7b502adee4de7e508ae8a7388aa59dddf70d24f366d65d010ee8R42-R46 we can [group ](https://docs.celeryproject.org/en/stable/reference/celery.html?highlight=chain#celery.group)these tasks:

```python
pre_register_tasks = group(
  register_appointment_task,
  generate_report_task
)
post_register_tasks = group(
  update_appointment_task,
  finalize_completion_task
)

chain(
  pre_register_tasks,

  register_submission_task,
  obtain_submission_reference_task,

  post_register_tasks 
)
```

**Comment by sergei-maertens** [comment]
Note: we should we generate a report already if the appointment registration fails because of user-recoverable error? It doesn't sit in the way, but it may be confusing.

We could also add an `on_error` handler to the appointment task to delete possible generated reports, but that's a race condition again in itself...

**Comment by sergei-maertens** [comment]
We're not running into issues with the way we currently do things and the entire chain was reworked with multi-step registration. Closing this as wontfix.
